tempesta_fw/ss_skb.c

```diff
@@ -1010,7 +1016,7 @@ ss_skb_queue_coalesce_tail(SsSkbList *skb_list, const struct sk_buff *skb)
 	if (headlen) {
 		BUG_ON(!skb->head_frag);
 		head_frag.size = headlen;
-		head_frag.page.p = virt_to_head_page(skb->head);
+		head_frag.page.p = virt_to_page(skb->head);
```

Comment:
I don't think there's a difference, but nevertheless, what is the reason for the change? These are definitely SKBs that are allocated by Tempesta to hold data from the original SKB. That means these SKBs are allocated by Tempesta's SKB allocator in the kernel, where SKB structure itself and the head are page fragments, and they are addressed exactly this way: https://github.com/tempesta-tech/linux-4.9.35-tfw/blob/master/net/core/skbuff.c#L489.

Is there a case where `skb->head` is not a head page? Perhaps, a better comment/explanation is in order?

Reply:
@keshonok thank you for the notes. Strictly speaking `skb->head` can be a tail page, so here we'd get wrong page. However, since we set offset, we set correct offset from the head page and everything works fine. However, the code is misleading and bit-bit slower - just has no sense to keep the bad code.